### ISSUE TITLE
Unreachable code should be detected by default without `--warn-unreachable` flag

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20264]
With [--warn-unreachable](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-warn-unreachable) flag, unreachable code can be detected as shown below:

```python
def func() -> int:
    return 100
    print('Unreachable')
# error: Statement is unreachable
```

But without `--warn-unreachable` flag, unreachable code cannot be detected as shown below so unreachable code should be detected by default without `--warn-unreachable` flag:

```python
def func() -> int:
    return 100
    print('Unreachable')
# No error
```

Actually, Java detects unreachable code by default as shown below:

```java
class Main {
    public static void main(String[] args) {
        System.out.println(func());
    }
    
    public static int func() {
        return 100;
        System.out.println("Unreachable"); // error: unreachable statement
    }
}
```

And, Go also detects unreachable code by default as shown below:

```golang
package main

import "fmt"

func main() {
	fmt.Println(myfunc())
}

func myfunc() int {
	return 100
	fmt.Println("Unreachable") // missing return
}
```

**Comment by A5rocks** [comment]
See https://github.com/python/mypy/issues/18078 for the enabling it by default with `--strict`. Probably it's a bad idea to enable it by default for non-strict! Java and Go have the advantages that the types *must* be followed... mypy doesn't have that. `--warn-unreachable` is an especially egregious false-positive in this case (that the type hints aren't followed somewhere in the program) so IMO it's bad UX to enable it by default.

I won't close this though.

**Comment by hauntsaninja** [comment]
Closing as a duplicate of https://github.com/python/mypy/issues/18078
